sunpy/database/tests/test_database.py

```diff
@@ -441,6 +445,52 @@ def test_add_entry_from_hek_qr(database):
     assert len(database) > 1
 
 
+@pytest.mark.remote_data
+def test_hek_query_download(monkeypatch, database, tmpdir):
+
+    assert len(database) == 0
+
+    records = ['94_1331820530-1331820530', '94_1331820542-1331820542',
+               '94_1331820554-1331820554', '94_1331820566-1331820566',
+               '94_1331820578-1331820578', '94_1331820590-1331820590',
+               '94_1331820602-1331820602', '94_1331820614-1331820614',
+               '94_1331820626-1331820626', '94_1331820638-1331820638']
+
+    def mock_parfive_download(obj, *args, **kwargs):
+
+        assert obj.http_queue.qsize() == 10
+        assert obj.ftp_queue.qsize() == 0
+
+        queue = obj.http_queue
+        obj_records = []
+
+        while not queue.empty():
+            url = queue.get_nowait().keywords['url']
+            obj_records.append(url[-24:])
+
+        assert obj_records == records
+
+        result = Results()
+        result.append(str(tmpdir))
+        return result
+
+    def mock_entries_from_dir(*args, **kwargs):
+        for i in range(10):
+            yield DatabaseEntry()
+
+    monkeypatch.setattr(Downloader, "download", mock_parfive_download)
+    monkeypatch.setattr(tables, "entries_from_dir", mock_entries_from_dir)
+
+    query = hek.HEKClient().search(
+        hek.attrs.Time('2019/03/10 14:40:10', '2019/04/11 16:40:50'),
+        hek.attrs.EventType('FL')
+    )
+
+    database.download_from_hek_query_result(query[4], path=str(tmpdir))
+
+    assert len(database) > 0 and len(database) == 10
```

Comment:
Having both checks seems redundant?

Reply:
You mean checking queue length and queue data or checking length of database at the end ?

Reply:
Yeah.

Reply:
So should I remove the check for length of database ?

Reply:
Just have one of them?

Reply:
Yeah sure :`)